src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -1259,7 +1259,38 @@ private void RunPriorityBuildTest(ProcessPriorityClass expectedPrority, params s
             logContents.ShouldContain(expected, () => logContents);
         }
 
-#region IgnoreProjectExtensionTests
+        /// <summary>
+        /// Test the case where there is one solution file, one solution filter file, and one project file.
+        /// </summary>
+        [Fact]
+        public void TestOneProjectOneSolutionFileOneSolutionFilterFile()
+        {
+            string[] projects = new string[] { "my.proj", "my.sln", "my.slnf" };
+            string[] extensionsToIgnore = Array.Empty<string>();
+            IgnoreProjectExtensionsHelper projectHelper = new IgnoreProjectExtensionsHelper(projects);
+            MSBuildApp.ProcessProjectSwitch(Array.Empty<string>(), extensionsToIgnore, projectHelper.GetFiles).ShouldBe("my.sln", StringCompareShould.IgnoreCase);
+        }
+
+        /// <summary>
+        /// Test the default file to build in cases involving at least one solution filter file.
+        /// </summary>
+        [Theory]
+        [InlineData("my.proj my.sln my.slnf my.sln")]
+        [InlineData("abc.proj bcd.csproj slnf.slnf other.slnf abc.proj")]
+        [InlineData("abc.sln slnf.slnf abc.slnf abc.sln")]
+        [InlineData("abc.csproj abc.slnf not.slnf abc.csproj")]
+        [InlineData("abc.slnf abc.slnf")]
+        public void TestDefaultBuildWithSolutionFilter(string projectData)
+        {
+            string[] projects = projectData.Split(' ');
+            string answer = projects[projects.Length - 1];
+            projects = projects.Take(projects.Length - 1).ToArray();
```

Comment:
Fun new (C# 8.0) language feature for you:

```suggestion
            string answer = projects[^1];
            projects = projects[..^1].ToArray();
```

[example](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQiAM7EGYAEA3uvqfogkSWTQMIAWAhmANoC6+ANgJYDOw+ALxEA5IxEAafCIBGk6QGMRAXwDc1GqThIAnAAoe/FgD0kbAJTq0mstv2HgLAHRPTFqzWXplQA=)

Reply:
Cool feature! VS doesn't seem to like it:
![image](https://user-images.githubusercontent.com/12969783/93633876-820c0500-f9a4-11ea-8208-f731728caef3.png)
Does that mean I need to add something to the csproj?

Reply:
Looks like we'd need a reference to `System.Runtime` since it's built in only on .NET Core 3+: https://docs.microsoft.com/dotnet/api/system.index

So leave it I guess. Sorry to mislead!